tools/wptrunner/wptrunner/executors/base.py

```diff
@@ -286,9 +286,9 @@ def run_test(self, test):
         """Run a particular test.
 
         :param test: The test to run"""
-        if test.environment != self.last_environment:
-            self.on_environment_change(test.environment)
         try:
+            if test.environment != self.last_environment:
```

Comment:
I wonder if this should be in a separate try/except block before this one, and perhaps results in an "INTERNAL-ERROR" or something? Maybe @jgraham can comment.

Reply:
Luke, if this is due to an issue at browser side, then "ERROR" is more appropriate? I tried and did not reproduce this on clank. What does "INTERNAL-ERROR" mean? thanks!

Reply:
I think `result_from_exception` will convert this into `INTERNAL-ERROR` and I think that's the correct result for this case (since it's an error not assoicated with the specific test code).